core/src/main/scala/com/nvidia/spark/rapids/tool/profiling/Analysis.scala

```diff
@@ -172,15 +172,12 @@ class Analysis(apps: Seq[ApplicationInfo]) {
   def sqlMetricsAggregation(): Seq[SQLTaskAggMetricsProfileResult] = {
     val allRows = apps.flatMap { app =>
       app.sqlIdToInfo.map { case (sqlId, sqlCase) =>
-        val jcs = app.jobIdToInfo.filter { case (_, jc) =>
-          jc.sqlID.isDefined && jc.sqlID.get == sqlId
-        }
-        if (jcs.isEmpty) {
+        val stagesInSQL = app.sqlIdToStages.getOrElse(sqlId, ArrayBuffer.empty)
```

Comment:
`app.sqlIdToStages.getOrElse(sqlId, ArrayBuffer.empty)`
I am not big fan of using `getOrElse` here because this would create a reference or an object which adds strength on the GC.
I have removed those `getOrElse` calls in previous PR for optimization reasons . So, we want to make sure that we do not reintroduce them back.

A more optimized version would be:
```
if (app.sqlIdToStages.contains(sqlId)) {
  // main code here
} else {
  NONE
}
```